python/ray/train/torch.py

```diff
@@ -289,52 +231,18 @@ def prepare_data_loader(
         move_to_device (bool): If set, automatically move the data
             returned by the data loader to the correct device.
     """
+    return get_session().accelerator.prepare_data_loader(
+        data_loader, add_dist_sampler=add_dist_sampler, move_to_device=move_to_device
+    )
 
-    # Only add Distributed Sampler if the following conditions hold:
-    # 1. More than one training worker is being used.
-    # 2. A DistributedSampler has not already been added by the user.
-    # 3. The dataset is not an IterableDataset. Samplers do not worker with
-    # IterableDatasets.
-    if (
-        train.world_size() > 1
-        and not isinstance(data_loader.sampler, DistributedSampler)
-        and not (
-            hasattr(data_loader, "dataset")
-            and isinstance(data_loader.dataset, IterableDataset)
-        )
-        and add_dist_sampler
-    ):
-
-        def with_sampler(loader):
-            # Automatically set the DistributedSampler
-
-            # If using a sampler, the shuffle attribute in the
-            # DataLoader must be set to False.
-            # Instead the shuffling is determined by the shuffle attribute
-            # in the DistributedSampler.
-            # We identify if shuffling is enabled in the passed in
-            # DataLoader by seeing if the sampler for the DataLoader is a
-            # SequentialSampler.
-            shuffle = not isinstance(loader.sampler, SequentialSampler)
-
-            data_loader_args = {
-                "dataset": loader.dataset,
-                "batch_size": loader.batch_size,
-                "shuffle": False,
-                "num_workers": loader.num_workers,
-                "collate_fn": loader.collate_fn,
-                "pin_memory": loader.pin_memory,
-                "drop_last": loader.drop_last,
-                "timeout": loader.timeout,
-                "worker_init_fn": loader.worker_init_fn,
-                "sampler": DistributedSampler(loader.dataset, shuffle=shuffle),
-            }
-            return DataLoader(**data_loader_args)
-
-        data_loader = with_sampler(data_loader)
-
-    if move_to_device:
-        device = get_device()
-        data_loader = _WrappedDataLoader(data_loader, device)
-
-    return data_loader
+
+@PublicAPI(stability="beta")
+def get_device() -> torch.device:
+    """Gets the correct torch device to use for training."""
+    return get_session().accelerator.get_device()
+
+
+@PublicAPI(stability="beta")
+def accelerate() -> None:
+    """Enables training optimizations."""
+    get_session().accelerator = TorchAccelerator()
```

Comment:
Random thought -  would it make sense to have 1 `accelerator_factory` `Callable` which is used to create the `Accelerator` here, and then the "default" `Accelerator` is just the case when `accelerator_factory` is called with default args?

That way there is only 1 way to create `Accelerator`

Reply:
I think that implementation has a couple of disadvantages:
- It would couple `TorchConfig.accelerator_factory` with `train.torch.accelerate`. Currently, `TorchConfig.default_accelerator_factory` must accept zero arguments. If we call a factory method with arguments in `train.torch.accelerate`, then `TorchConfig.accelerator_factory` would need to have a specific function signature to be compatible with `train.torch.accelerate`.
- It introduces a layer of indirection, because you would pass arguments to the factory which would pass arguments to the constructor, rather than just passing arguments to the constructor directly.

Reply:
Updated the implementation to use a class instead of a factory.